src/Service/ProductBundleManipulator.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * Created by solutionDrive GmbH
+ *
+ * @copyright 2018 solutionDrive GmbH
+ */
+
+namespace solutionDrive\SyliusProductBundlesPlugin\Service;
+
+use solutionDrive\SyliusProductBundlesPlugin\Entity\ProductBundleInterface;
+use solutionDrive\SyliusProductBundlesPlugin\Entity\ProductBundleSlotInterface;
+use solutionDrive\SyliusProductBundlesPlugin\Service\Options\ProductBundleSlotOptionsInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+
+class ProductBundleManipulator implements ProductBundleManipulatorInterface
+{
+    /** @var FactoryInterface */
+    private $productBundleSlotFactory;
+
+    /** @var ProductBundleInterface */
+    private $productBundle;
+
+    public function __construct(FactoryInterface $productBundleSlotFactory)
+    {
+        $this->productBundleSlotFactory = $productBundleSlotFactory;
+    }
+
+    public function setProductBundle(ProductBundleInterface $productBundle): void
+    {
+        $this->productBundle = $productBundle;
+    }
+
+    public function getProductBundle(): ?ProductBundleInterface
+    {
+        return $this->productBundle;
+    }
+
+    /**
+     * @param ProductInterface[] $products
+     */
+    public function addSlot(
+        string $slotName,
+        ProductBundleSlotOptionsInterface $options = null,
```

Comment:
Im wondering that this works without declaring the parameter as nullable

Reply:
maybe there is a check missing in ecs?

Would it be better like:
```php
public function addSlot(
        string $slotName,
        ?ProductBundleSlotOptionsInterface $options = null,
        array $products = []
    ): void;
```